django/http/response.py

```diff
@@ -320,6 +320,7 @@ class HttpResponse(HttpResponseBase):
     """
 
     streaming = False
+    non_pickable_attrs = ['client', 'templates', 'context', 'json', 'resolver_match', 'wsgi_request']
```

Comment:
this attribute should be `non_picklable_attrs` I think.

Reply:
Actually, I think it might be useful if there were a comment preceding this line to say which attributes are only being dropped because of the test `Client` implementation. eg: I think `json` is test client specific, rather than `HttpResponse` related?

What's the _minimum_ set of values required to dump an HttpResponse _without_ using the test client? (ie: using `RequestFactory` for a view and then doing `pickle.dumps(response)`? 

@felixxm Design decision needed. Do we want the list to contain values which are required for the test client only, so that responses pickled during unit tests work correctly, _or_ do we want the minimum required for dumping of 'normal' responses which haven't been massaged by `Client.request` et al?

Reply:
I think it's fine to also include non-pickleable attributes added by test clients. I'd add a comment about them, e.g.
```
non_pickleable_attrs = frozenset([
    'resolver_match',
    # Non-pickleable attributes added by test clients.
    'asgi_request', 'client', 'context', 'json', 'templates', 'wsgi_request',
])
```


It seems that `asgi_request` is missing in your list :thinking: Can we add a test for the async-client?

> this attribute should be `non_picklable_attrs` I think.

... or `pickleable` :smile: I need to confirm with Carlton.

Reply:
> > this attribute should be `non_picklable_attrs` I think.
> 
> ... or `pickleable` 😄 I need to confirm with Carlton.

Yeah that's how I was going to recommend spelling it, because it _felt/looked_ right, but then I looked at the python pickle docs and they use `picklable` everywhere. TIL.

Reply:
We're more consistent then Python when it comes to "pickl(e)able":
- Python: `picklable` - 92, `pickleable` - 39
- Django: `picklable` - 45, `pickleable` - 1

Reply:
I _think_ the 'e' should be dropped, and more so in `en-US` than 'british' English.